packages/router/router.ts

```diff
@@ -400,7 +405,7 @@ export interface StaticHandler {
   dataRoutes: AgnosticDataRouteObject[];
   query(
     request: Request,
-    opts?: { requestContext?: unknown }
+    opts?: { loadRouteIds?: string[]; requestContext?: unknown }
```

Comment:
`loadRouteIds` is used by single fetch for fine-grained revalidation, so we can run `query` but only load a subset of matched loaders